reference/opt/shaders-hlsl/comp/shared.comp

```diff
@@ -18,6 +18,12 @@ void comp_main()
     sShared[gl_LocalInvocationIndex] = asfloat(_22.Load(gl_GlobalInvocationID.x * 4 + 0));
     GroupMemoryBarrierWithGroupSync();
     _44.Store(gl_GlobalInvocationID.x * 4 + 0, asuint(sShared[(4u - gl_LocalInvocationIndex) - 1u]));
+    GroupMemoryBarrierWithGroupSync();
```

Comment:
6 barriers back to back here is pretty intense, this seems like overkill.

Reply:
I was just making sure each GLSL intrinsic was translated into the expected HLSL intrinsic. In general, I think I'd feel safer if SPIRV-Cross' tests eventually covered every GLSL intrinsic. If this is too much, some of the memoryBarrier*() and the intrinsics that were already tested a few lines up could probably be removed.

Reply:
right, ok, we should have tests for all possible combinations.